### PR TITLE
Implemented Wallet Table component

### LINKED ISSUE #133: Implement `WalletTable` component
### Goal

Implemented `WalletTable` component in the storybook

### Context

For the `Bidding with DAI` feature, we need to display some information about connected wallet in a structured way. For the names of the props, please refer to the ([already implemented](https://github.com/sidestream-tech/unified-auctions-ui/pull/127)) `DepositBlock` component, which displays some of the same values.

Please note: 
- in case there is no wallet, the table will not be displayed, therefore, wallet-related props should be `required`
- last time updated is a date when the wallet data was last time fetched
- `refresh` just emits an event to refetch the data

### Assets
MakerDAO@2x (27)

- Bidding with DAI mock: https://github.com/sidestream-tech/unified-auctions-ui/issues/117

### Tasks
- [ ] create new `WalletTable` inside new `wallet` folder
- [ ] create stories
- [ ] please check that `isLoading` is taken into the account

### PR DESCRIPTION
**Pull request by zoey-kaiser**
closes #133 

Normal State:
Screenshot 2022-03-10 at 15 37 21

Loading State:
Screenshot 2022-03-10 at 15 37 24

### CODE REVIEW

File: frontend/components/wallet/WalletTable.stories.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    methods: {
+        refresh: action('refresh'),
+    },
+};
+
+storiesOf('Wallet/WalletTable', module)
+    .add('Default', () => ({
+        ...common,
+        template:
+            '<WalletTable :walletAddress="walletAddress" :walletETH="walletETH" :walletDAI="walletDAI" :walletVatDAI="walletVatDAI" :walletLastUpdatedDate="walletLastUpdatedDate" @refresh="refresh" />',
```
Comment (valiafetisov) [inline frontend/components/wallet/WalletTable.stories.js:25]: Did you know that you can v-bind an object pass all of it's keys as props?
```suggestion
            '<WalletTable v-bind="$data" @refresh="refresh" />',
```

File: frontend/components/wallet/WalletTable.vue [hunk 2]
```diff
@@ -0,0 +1,92 @@
+<template>
+    <Loading :is-loading="isLoading">
+        <table class="w-full table-fixed border-collapse border">
+            <tbody>
+                <tr>
+                    <td>Wallet Address</td>
+                    <td>
+                        <FormatAddress :value="walletAddress" :shorten="true" />
```
[the remark this answers is not in the document]
Comment (benediktkaiser) [reply frontend/components/wallet/WalletTable.vue:8]: It currently is. Do you want me to not make it a link?

File: frontend/components/wallet/WalletTable.vue [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                    <td>Currently deposited in VAT</td>
+                    <td>
+                        <FormatCurrency :value="walletVatDAI" currency="DAI" />
+                    </td>
+                </tr>
+                <tr>
+                    <td>Last updated</td>
+                    <td>
+                        <TimeTill :date="walletLastUpdatedDate" />
+                        <button class="text-green-600 hover:text-green-400 underline ml-1" @click="$emit('refresh')">
```
[the remark this answers is not in the document]
Comment (benediktkaiser) [reply frontend/components/wallet/WalletTable.vue:33]: I created a new Type. I had to do some refactoring of the button component for this. Stories for all Button Types can be found!

File: frontend/components/wallet/WalletTable.stories.js (hunk 1 above)
Comment (benediktkaiser) [reply frontend/components/wallet/WalletTable.stories.js:25]: Cool. I did not know this!

File: frontend/components/wallet/WalletTable.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (valiafetisov) [reply frontend/components/wallet/WalletTable.vue:8]: No, sorry, it's just on the screenshot it didn't look like a link. Is it a style problem? Do we need to wrap it into a TextBlock to apply correct link style?

[the remark this answers is not in the document]
Comment (benediktkaiser) [reply frontend/components/wallet/WalletTable.vue:8]: True. I will add the TextBlock.

**Review by valiafetisov — DISMISSED** [review]

**Review by LukSteib — CHANGES_REQUESTED** [review]
- [ ] The wallet link used in story redirects to an etherscan url with path `.../tx/...`. To properly link to a wallet we need `.../address/...`. Or is this just due to the faker we are using for storybook creation and will be handled correclty once an actual address is passed here?
- [ ] Have we thought about the following cases for the table 
   - [ ] what to display if no wallet is connected
   - [ ] what to display if wallet is connect but VAT is not authorised (are we able to get information for table row `Currently deposited in VAT` in this case?)

**Comment by valiafetisov** [comment]
> what to display if no wallet is connected

No table will be displayed in this case, but the WalletBlock

> are we able to get information for table row Currently deposited in VAT in this case?

Yes, since blockchain is public, knowing a balance doesn't require additional rights

**Comment by valiafetisov** [comment]
Actually, sorry that I'm changing my mind @benediktkaiser, but let's account for no walletAddress within this component, since we have `isLoading` which can't be used otherwise
- [ ] please make props optional
- [ ] please show WalletBlock if no `walletAddress`
- [ ] please show `Unknown` is other bignumber props are undefined (but not 0)
- [ ] please handle no date as `Unknown` as well

**Comment by benediktkaiser** [comment]
> * [ ]  please show `Unknown` is other bignumber props are undefined (but not 0)

We are only accepting BigNumbers here meaning that the props will never just be 0, as 0 displayed as a BigNumber is an object. Therefore I only need to see if the prop is undefined.

**Comment by benediktkaiser** [comment]
Screenshot 2022-03-11 at 14 46 57

**Review by LukSteib — DISMISSED** [review]

**Comment by valiafetisov** [comment]
> - [x] please show WalletBlock if no `walletAddress`

I don't see that you've used `WalletBlock`

**Comment by benediktkaiser** [comment]
> I don't see that you've used `WalletBlock`. The original suggestion, if it wasn't clear: if there is no `walletAddress`, it means no wallet is connected and instead of showing a table, we can show the `WalletBlock` component which contains a button to connect a wallet. Or do you have another suggestion how to handle "no connected wallet" state in this component?

That makes sense. Sorry I completely misunderstood it. I like the idea a lot and implemented it!

File: frontend/components/wallet/WalletTable.vue [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            default: null,
+        },
+        walletVatDAI: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        walletLastUpdatedDate: {
+            type: [String, Number, Date],
+            default: null,
+        },
```
Comment (valiafetisov) [inline frontend/components/wallet/WalletTable.vue:94]: (I actually forgot to implement that one in #143)

**Review by valiafetisov — APPROVED** [review]

**Review by LukSteib — APPROVED** [review]

**Pull request merged**